apps/core/src/hooks/useGetValidatorsEvents.ts

```diff
@@ -2,33 +2,56 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { useRpcClient } from '../api/RpcClientContext';
-import { type EventId, VALIDATORS_EVENTS_QUERY } from '@mysten/sui.js';
+import {
+    type EventId,
+    VALIDATORS_EVENTS_QUERY,
+    SuiEvent,
+} from '@mysten/sui.js';
 import { useQuery } from '@tanstack/react-query';
 
 type GetValidatorsEvent = {
-    cursor?: EventId | null;
     limit: number | null;
     order: 'ascending' | 'descending';
 };
 
+// NOTE: This copys the query limit from our Rust JSON RPC backend, this needs to be kept in sync!
+const QUERY_MAX_RESULT_LIMIT = 1000;
+
 //TODO: get validatorEvents by validator address
-export function useGetValidatorsEvents({
-    cursor,
-    limit,
-    order,
-}: GetValidatorsEvent) {
+export function useGetValidatorsEvents({ limit, order }: GetValidatorsEvent) {
     const rpc = useRpcClient();
-    // since we are getting events base on the number of validators, we need to make sure that the limit is not null and cache by the limit
-    // number of validators can change from network to network
+    // Since we are getting events based on the number of validators, we need to make sure that the limit
+    // is not null and cache by the limit number of validators can change from network to network
     return useQuery(
-        ['validatorEvents', limit, cursor, order],
-        () =>
-            rpc.queryEvents({
+        ['validatorEvents', limit, order],
+        async () => {
+            if (limit! > QUERY_MAX_RESULT_LIMIT) {
+                let hasNextPage = true;
+                let currCursor: EventId | null | undefined;
+                const results: SuiEvent[] = [];
+
+                while (hasNextPage && results.length < limit!) {
+                    const validatorEventsResponse = await rpc.queryEvents({
+                        query: { MoveEventType: VALIDATORS_EVENTS_QUERY },
+                        cursor: currCursor,
+                        limit: Math.min(limit!, QUERY_MAX_RESULT_LIMIT),
+                        order,
+                    });
+
+                    hasNextPage = validatorEventsResponse.hasNextPage;
+                    currCursor = validatorEventsResponse.nextCursor;
+                    results.push(...validatorEventsResponse.data);
+                }
+                return results;
+            }
```

Comment:
One final note here is that this API can now return more than the limit. I'd truncate the limit at the end just to avoid that.